consensus/hotstuff/vote_collector.go

```diff
@@ -14,15 +14,13 @@ type VoteCollector interface {
 	// return error if the signature is invalid
 	// When enough votes have been added to produce a QC, the QC will be created asynchronously, and
 	// passed to EventLoop through a callback.
+	// ToDo: document sentinel errors expected during normal operation
 	AddVote(vote *model.Vote) (bool, error)
 
-	// VoteCreator returns a function that is able to create a vote for a given block
-	// The returned createVote function holds a stateful crypto signer object that is able to
-	// verify and store a signature share from other nodes, as well as produce a signature for
-	// our own vote. It is used by Voter to create our own vote.
-	VoteCreator() createVote
-
-	// BlockID returns the block ID that the collector is collecting votes for.
+	// BlockID returns the block ID that this instance is collecting votes for.
 	// This method is useful when adding the newly created vote collector to vote collectors map.
 	BlockID() flow.Identifier
+
+	// Block returns the block that this instance is collecting votes for.
+	Block() *model.Block
```

Comment:
I think it's too much, if we think of a case where we have starting collecting votes but still haven't received a block, then we can't return it. Maximum what we can get is `BlockID` and `View`

Reply:
👍